stablehlo/tests/interpret_sign.mlir

```diff
@@ -0,0 +1,168 @@
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si64
+func.func @sign_op_test_si64() -> tensor<3xi64> {
+  %operand = stablehlo.constant dense<[-9223372036854775807, 0, 9223372036854775807]> : tensor<3xi64>
+  %result = stablehlo.sign %operand : tensor<3xi64>
+  func.return %result : tensor<3xi64>
+  // CHECK-NEXT: tensor<3xi64>
+  // CHECK-NEXT: -1 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 1 : i64
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_bf16
+func.func @sign_op_test_bf16() -> tensor<8xbf16> {
+  // -0.0, +0.0, -3.14, +3.14, -NaN, +NaN, -Inf, +Inf
+  %operand = stablehlo.constant dense<[-0.0, 0.0, -3.140625, 3.140625, 0xFFFF, 0x7FFF, 0xFF80, 0x7F80]> : tensor<8xbf16>
+  %result = stablehlo.sign %operand : tensor<8xbf16>
+  func.return %result : tensor<8xbf16>
+  // CHECK-NEXT: tensor<8xbf16>
+  // CHECK-NEXT: -0.000000e+00 : bf16
+  // CHECK-NEXT: 0.000000e+00 : bf16
+  // CHECK-NEXT: -1.000000e+00 : bf16
+  // CHECK-NEXT: 1.000000e+00 : bf16
+  // CHECK-NEXT: 0xFFFF : bf16
+  // CHECK-NEXT: 0x7FFF : bf16
+  // CHECK-NEXT: -1.000000e+00 : bf16
+  // CHECK-NEXT: 1.000000e+00 : bf16
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_f16
+func.func @sign_op_test_f16() -> tensor<6xf16> {
+  // -0.0, +0.0, -3.14, +3.14, -NaN, +NaN, -Inf, +Inf
+  %operand = stablehlo.constant dense<[-0.0, 0.0, -3.141, 3.141, 0xFFFF, 0x7FFF]> : tensor<6xf16>
+  %result = stablehlo.sign %operand : tensor<6xf16>
+  func.return %result : tensor<6xf16>
+  // CHECK-NEXT: tensor<6xf16>
+  // CHECK-NEXT: -0.000000e+00 : f16
+  // CHECK-NEXT: 0.000000e+00 : f16
+  // CHECK-NEXT: -1.000000e+00 : f16
+  // CHECK-NEXT: 1.000000e+00 : f16
+  // CHECK-NEXT: 0xFFFF : f16
+  // CHECK-NEXT: 0x7FFF : f16
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_f32
+func.func @sign_op_test_f32() -> tensor<8xf32> {
+  // -0.0, +0.0, -3.14, +3.14, -NaN, +NaN, -Inf, +Inf
+  %operand = stablehlo.constant dense<[-0.0, 0.0, -3.14159265, 3.14159265, 0xFFFFFFFF, 0x7FFFFFFF, 0xFF800000, 0x7F800000]> : tensor<8xf32>
+  %result = stablehlo.sign %operand : tensor<8xf32>
+  func.return %result : tensor<8xf32>
+  // CHECK-NEXT: tensor<8xf32>
+  // CHECK-NEXT: -0.000000e+00 : f32
+  // CHECK-NEXT: 0.000000e+00 : f32
+  // CHECK-NEXT: -1.000000e+00 : f32
+  // CHECK-NEXT: 1.000000e+00 : f32
+  // CHECK-NEXT: 0xFFFFFFFF : f32
+  // CHECK-NEXT: 0x7FFFFFFF : f32
+  // CHECK-NEXT: -1.000000e+00 : f32
+  // CHECK-NEXT: 1.000000e+00 : f32
+}
+
+// -----
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_f64
+func.func @sign_op_test_f64() -> tensor<8xf64> {
+  // -0.0, +0.0, -3.14, +3.14, -NaN, +NaN, -Inf, +Inf
+  %operand = stablehlo.constant dense<[-0.0, 0.0, -3.14159265358979323846, 3.14159265358979323846, 0xFFFFFFFFFFFFFFFF, 0x7FFFFFFFFFFFFFFF, 0xFFF0000000000000, 0x7FF0000000000000]> : tensor<8xf64>
```

Comment:
Let's test numbers which are relevant to the logic of the code without duplicating logical paths:
```
// +NaN, -1.0, -0.0, +0.0, 1.0
%operand = stablehlo.constant dense<[0x7FFFFFFFFFFFFFFF, -1.0, -0.0, 0.0, 1.0]> : tensor<5xf64>
```